libcudacxx/include/cuda/std/string_view

```diff
@@ -0,0 +1,800 @@
+
+#ifndef _CUDA_STD_STRING_VIEW
+#define _CUDA_STD_STRING_VIEW
+
+#include <cuda/std/detail/__config>
+
+#if defined(_CCCL_IMPLICIT_SYSTEM_HEADER_GCC)
+#  pragma GCC system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_CLANG)
+#  pragma clang system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_MSVC)
+#  pragma system_header
+#endif // no system header
+
+#include <cuda/std/__algorithm/min.h>
+#include <cuda/std/__cstddef/types.h>
+#include <cuda/std/__functional/hash.h>
+#include <cuda/std/__functional/unary_function.h>
+#include <cuda/std/__fwd/string_view.h>
+#include <cuda/std/__iterator/access.h>
+#include <cuda/std/__iterator/bounded_iter.h>
+#include <cuda/std/__iterator/concepts.h>
+#include <cuda/std/__iterator/data.h>
+#include <cuda/std/__iterator/empty.h>
+#include <cuda/std/__iterator/iterator_traits.h>
+#include <cuda/std/__iterator/reverse_access.h>
+#include <cuda/std/__iterator/reverse_iterator.h>
+#include <cuda/std/__iterator/size.h>
+#include <cuda/std/__memory/pointer_traits.h>
+#include <cuda/std/__ranges/concepts.h>
+#include <cuda/std/__ranges/data.h>
+#include <cuda/std/__ranges/enable_borrowed_range.h>
+#include <cuda/std/__ranges/enable_view.h>
+#include <cuda/std/__ranges/size.h>
+#include <cuda/std/__string/char_traits.h>
+#include <cuda/std/__string/helper_functions.h>
+#include <cuda/std/__type_traits/is_array.h>
+#include <cuda/std/__type_traits/is_convertible.h>
+#include <cuda/std/__type_traits/is_same.h>
+#include <cuda/std/__type_traits/is_standard_layout.h>
+#include <cuda/std/__type_traits/is_trivially_constructible.h>
+#include <cuda/std/__type_traits/is_trivially_copyable.h>
+#include <cuda/std/__type_traits/remove_cvref.h>
+#include <cuda/std/__type_traits/remove_reference.h>
+#include <cuda/std/__type_traits/type_identity.h>
+#include <cuda/std/detail/libcxx/include/stdexcept>
+#include <cuda/std/limits>
+#include <cuda/std/version>
+
+#if !_CCCL_COMPILER(NVRTC)
+#  include <iosfwd>
+#endif // !_CCCL_COMPILER(NVRTC)
+
+_CCCL_PUSH_MACROS
+
+_LIBCUDACXX_BEGIN_NAMESPACE_STD
+
+template <class _CharT, class _Traits>
+class basic_string_view
+{
+public:
+  using traits_type            = _Traits;
+  using value_type             = _CharT;
+  using pointer                = _CharT*;
+  using const_pointer          = const _CharT*;
+  using reference              = _CharT&;
+  using const_reference        = const _CharT&;
+  using const_iterator         = const_pointer;
+  using iterator               = const_iterator;
+  using const_reverse_iterator = _CUDA_VSTD::reverse_iterator<const_iterator>;
+  using reverse_iterator       = const_reverse_iterator;
+  using size_type              = size_t;
+  using difference_type        = ptrdiff_t;
+
+  static constexpr const size_type npos = static_cast<size_type>(-1);
+
+  static_assert(!_CCCL_TRAIT(is_array, value_type), "character type of basic_string_view must not be an array");
```

Comment:
Nitpick: I believe we should try and use the `_v` versions everywhere. I was unsure whether we want to avoid paying the cost of instantiating an inline variable for those that directly forward to a compiler builtin but that seems more and more overkill